src/OpenRoad.cc

```diff
@@ -507,9 +507,12 @@ void OpenRoad::writeDb(std::ostream& stream)
 
 void OpenRoad::writeDb(const char* filename)
 {
-  utl::StreamHandler stream_handler(filename, true);
-
-  db_->write(stream_handler.getStream());
+  try {
+    utl::StreamHandler stream_handler(filename, true);
+    db_->write(stream_handler.getStream());
+  } catch (const std::ios_base::failure& f) {
+    logger_->error(ORD, 55, "Cannot write database to {}", filename);
```

Comment:
Please include the f.what string in the message as it may give more detail on the cause of the failure.

Reply:
Yes for write_lef. write_def is different with the '.1'. Where does it come from?
```
>>> write_def foo/bar
[ERROR GUI-0070] No such file or directory: foo/bar.1
>>> write_lef foo/bar
[ERROR GUI-0070] basic_ios::clear: iostream error (failed to open 'foo/bar.1'): iostream error
```

Reply:
write_lef might need to write multiple files so they get .<num> suffixes.

write_lef/def are not part of GUI.

Reply:
What is the case where write_lef write multiple files?
Regarding write_db & write_def, why using a suffix then? Should we use basic iostream instead?